### PR TITLE
More work on PDF publishing (BL-11437)

### PR DESCRIPTION
**Pull request by StephenMcConnel**
This change is [Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608)

### CODE REVIEW

**Review by andrew-polk — CHANGES_REQUESTED** [review]
I would like to get @hatton's take on the general approach to tweaking the page sizes for pdf printing before we merge.

> Reviewed 5 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-:-NNCUzqs8Jly0djRx5zg:b-bjy9lq)* status: all files reviewed, 2 unresolved discussions (waiting on @StephenMcConnel)

___
*[`src/content/bookLayout/basePage.less` line 410 at r1](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-NNCUoel5Qzb0yKfwyJA:-NNCUoel5Qzb0yKfwyJB:bhvep4y) ([raw file](https://github.com/BloomBooks/BloomDesktop/blob/cc96776ab958ff3391c35980e78310cd8f216b7f/src/content/bookLayout/basePage.less#L410)):*
> ```Less
>         &.A4Portrait {
>             min-width: 210mm;
>             max-width: 210mm;
> ```

@A4Portrait-Width



_Code quote:_
```Less
210mm
```
___
*[`src/content/bookLayout/basePage.less` line 416 at r1](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-NNCUs5A0Dek3oRXxP0v:-NNCUs5A0Dek3oRXxP0w:bpf9v5y) ([raw file](https://github.com/BloomBooks/BloomDesktop/blob/cc96776ab958ff3391c35980e78310cd8f216b7f/src/content/bookLayout/basePage.less#L416)):*
> ```Less
>         &.A4Landscape {
>             min-width: 297mm;
>             max-width: 297mm;
> ```

@A4Landscape-Width



_Code quote:_
```Less
297mm
```

**Review by andrew-polk — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @StephenMcConnel)

**Review by andrew-polk — CHANGES_REQUESTED** [review]
> Reviewed 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608)* status: all files reviewed, 1 unresolved discussion (waiting on @StephenMcConnel)

___
*[`src/content/bookLayout/basePage.less` line 451 at r3](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-NNN3GXY8MEJWTPjHCv_:-NNN3GXY8MEJWTPjHCva:bm6u6or) ([raw file](https://github.com/BloomBooks/BloomDesktop/blob/b6a03f4eb070bbccbd3d5f8f32fcf8ff27775468/src/content/bookLayout/basePage.less#L451)):*
> ```Less
>             max-width: (floor((@Cm13Square-Side / 25.4) * 96) / 96) * 25.4;
>             min-height: (ceil((@Cm13Square-Side / 25.4) * 96) / 96) * 25.4;
>             max-height: (ceil((@Cm13Square-Side / 25.4) * 96) / 96) * 25.4;
> ```

I suppose all these calcs are done in the process of creating the .css file?
In other words, the user's machine won't do these calcs every time they are used, right?
If they will, we could (should?) just document the calculation and put the numbers in directly.

**Review by StephenMcConnel — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608)* status: all files reviewed, 1 unresolved discussion (waiting on @andrew-polk)

___
*[`src/content/bookLayout/basePage.less` line 451 at r3](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-NNN3GXY8MEJWTPjHCv_:-NNN8IIo94o_7YFJ5pMa:bi31b5h) ([raw file](https://github.com/BloomBooks/BloomDesktop/blob/b6a03f4eb070bbccbd3d5f8f32fcf8ff27775468/src/content/bookLayout/basePage.less#L451)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I suppose all these calcs are done in the process of creating the .css file?
In other words, the user's machine won't do these calcs every time they are used, right?
If they will, we could (should?) just document the calculation and put the numbers in directly.
</blockquote></details>

These calculations are done by the less compilation process, with the result stored in the css file.

**Review by andrew-polk — APPROVED** [review]
I'm going to merge for now. @hatton did comment in Zulip on the approach without rejecting it.
We can always return to this if needed.

> *[Reviewable](https://reviewable.io/reviews/BloomBooks/BloomDesktop/5608#-:-NNNB-ZC2sJ6SMokx0kr:bdnuro7)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @StephenMcConnel)
